src/utils/helpers.ts

```diff
@@ -57,3 +57,42 @@ export function getCoords({
 
   return new Matrix2D().translate(tx, ty).scale(zoom).inverse();
 }
+
+/**
+ * Given a nodeId to find, a list of nodes to check against, and an optional parentId of the node
+ * find the node from the list of nodes
+ */
+export function findNestedNode(
+  nodeId: string,
+  children: any[],
+  parentId?: string
+) {
+  if (!nodeId || !children) {
+    return;
+  }
+
+  let foundNode = children.find((c) => c.id === nodeId);
```

Comment:
```suggestion
  let foundNode = children.find(n => n.id === nodeId);
```

Reply:
You don't really use the assignment - u could make this a const and cleanup code below. Over assignment out of scope can be confusing.